src/components/drawer/drawer.component.js

```diff
@@ -0,0 +1,80 @@
+import React, { useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import Icon from '../icon';
+
+import {
+  StyledDrawerWrapper,
+  StyledDrawerContent,
+  StyledIconButton,
+  StyledDrawerChildren,
+  StyledDrawerSidebar
+} from './drawer.style';
+
+const Drawer = ({
+  defaultIsExpanded,
+  expanded,
+  onChange,
+  children,
+  expandedWidth,
+  sidebar,
+  ...props
+}) => {
+  const isControlled = expanded !== undefined;
+  const [isExpandedInternal, setIsExpandedInternal] = useState(defaultIsExpanded || false);
+  const [firstOpen, setFirstOpen] = useState(true);
+  const isExpanded = isControlled ? expanded : isExpandedInternal;
+  const toggleDrawer = useCallback((ev) => {
+    if (!isControlled) setIsExpandedInternal(!isExpanded);
+    if (onChange) onChange(ev, !isExpanded);
+    setFirstOpen(false);
+  }, [isControlled, isExpanded, onChange]);
+
+  const closedClass = firstOpen ? '' : 'closed';
+
+  return (
+    <StyledDrawerWrapper
+      isExpanded={ isExpanded }
+      data-element='drawer'
+      { ...props }
+    >
+      <StyledDrawerContent
+        expandedWidth={ expandedWidth }
+        className={ isExpanded ? 'open' : closedClass }
+      >
+        <StyledIconButton
+          data-element='drawer-toggle'
+          onAction={ toggleDrawer }
+        >
+          <Icon type='chevron_right' />
+        </StyledIconButton>
+        <StyledDrawerSidebar>
+          { sidebar }
+        </StyledDrawerSidebar>
+      </StyledDrawerContent>
+      <StyledDrawerChildren>
+        {children}
+      </StyledDrawerChildren>
+    </StyledDrawerWrapper>
+  );
+};
+
+Drawer.propTypes = {
+  children: PropTypes.node,
+  /** Set the default state of expansion of the Drawer if component is meant to be used as uncontrolled */
+  defaultIsExpanded: PropTypes.bool,
```

Comment:
This should match the controlled prop `expanded`.
```suggestion
  defaultExpanded: PropTypes.bool,
```

Reply:
fixed.